### PR TITLE
Feature/rpc revoke permissions

### PR DESCRIPTION
**Pull request by alaahd**
## Explanation
This pull request introduces a new JSON-RPC method, "wallet_revokePermissions," to the MetaMask extension. The method allows revoking permissions for connected accounts and dapps. It provides the flexibility to revoke an entire permission object, deleting all accounts connected to a specific invoker, or to update permissions for a specific invoker by revoking a single connected account or a list of connected accounts.

## Changes Made
- Implemented the "wallet_revokePermissions" method in the relevant module.
- Updated the JSON-RPC interface to include the new method.
- Modified the permission controller to handle the revocation of accounts or permissions.
- Added the `revokePermissions` function responsible for revoking specific accounts or all accounts connected to a given origin.
- Created the `removeAllAccountPermissions` function to remove the specified account from all `eth_accounts` permissions and revoke the permission entirely if it only included the specified account.
- Updated the relevant files to import and utilize the newly added functions.

## Documentation
- Updated the documentation to include details about the new "wallet_revokePermissions" method.
- Documented the purpose, parameters, and expected response of the method.
- Provided usage examples and instructions for developers integrating with the MetaMask extension.

### After

https://github.com/MetaMask/metamask-extension/assets/2320312/97bc288e-5422-4871-a7fe-abed76acf10e

## Pre-merge author checklist

- [ ] I've clearly explained:
  - [ ] What problem this PR is solving
  - [ ] How this problem was solved
  - [ ] How reviewers can test my changes
- [ ] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

File: app/scripts/lib/rpc-method-middleware/handlers/revoke-permissions.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  end,
+  { handleRevokePermissions },
+) {
+  try {
+    const { permission } = req.params;
+    res.result = await handleRevokePermissions(permission);
+
+    return end();
+  } catch (error) {
+    if (error.message === 'User rejected to watch the asset.') {
```
Comment (GuiBibeau) [inline app/scripts/lib/rpc-method-middleware/handlers/revoke-permissions.js:44]: seems like this is the wrong error

Comment (julesat22) [reply app/scripts/lib/rpc-method-middleware/handlers/revoke-permissions.js]: Yeah, i've removed this error altogether as it isn't relevant to this endpoint

**Comment by julesat22** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by julesat22** [comment]
This PR aims to add the functionality for revoking permissions and is now re-opened for review. The feature has been locally tested and is ready for integration into the Portfolio. The endpoint has been validated through local testing and using the API playground.

**Review by GuiBibeau — APPROVED** [review]
Awesome improvements towards DX for dapp developers and for UX for dapp users!

**Comment by julesat22** [comment]
Closing - new PR has bene opened it its place: https://github.com/MetaMask/metamask-extension/pull/21216

**Pull request closed without merging**